tasks/actions.yaml

```diff
@@ -0,0 +1,176 @@
+variables:
+  - name: REGISTRY1_USERNAME
+    default: ""
+  - name: REGISTRY1_PASSWORD
+    default: ""
+  - name: GH_TOKEN
+    default: ""
+  - name: CHAINGUARD_IDENTITY
+    default: ""
+  - name: CHART_PATH
+    default: chart/
+  - name: GROUP_NAME
+    default: package
+  - name: COMMON_ZARF
+    default: "true"
+  - name: TYPE
+    default: ""
+  - name: FLAVOR
+    default: ""
+  - name: OPTIONS
+    default: ""
+  - name: GO_VERSION
+    default: 1.23.1
+
+tasks:
+  - name: debug-output
+    actions:
+      - description: Print basic debug info for a k8s cluster
+        cmd: |
+          echo "::group::kubectl get all"
+          uds zarf tools kubectl get all -A || echo "failed" | tee /tmp/debug-k-get-all.log
+          echo "::endgroup::"
+          echo "::group::kubectl get pv,pvc"
+          uds zarf tools kubectl get pv,pvc -A || echo "failed" | tee /tmp/debug-k-get-pv-pvc.log
+          echo "::endgroup::"
+          echo "::group::kubectl get package"
+          uds zarf tools kubectl get package -A || echo "failed" | tee /tmp/debug-k-get-package.log
+          echo "::endgroup::"
+          echo "::group::kubectl get events"
+          uds zarf tools kubectl get events -A --sort-by='.lastTimestamp' || echo "failed" | tee /tmp/debug-k-get-events.log
+          echo "::endgroup::"
+          echo "::group::kubectl describe nodes"
+          uds zarf tools kubectl describe nodes k3d-uds-server-0 || echo "failed" | tee /tmp/debug-k-describe-node.log
+          echo "::endgroup::"
+
+  - name: clean-gh-runner
+    actions:
+      - description: Clean up space on the GitHub runner
+        cmd: |
+            df -h
+            sudo rm -rf /usr/share/dotnet
+            sudo rm -rf /usr/local/lib/android
+            sudo rm -rf /opt/ghc
+            sudo rm -rf /opt/hostedtoolcache/CodeQL
+            sudo docker image prune --all --force
+            df -h
+
+  - name: save-logs
+    actions:
+      - description: Pull logs from containerd
+        cmd: |
+          CONTAINER_NAME="k3d-uds-server-0"
+          if docker ps | grep -q "$CONTAINER_NAME"; then
+            echo "Container $CONTAINER_NAME is running. Proceeding with log copy..."
+            docker cp ${CONTAINER_NAME}:/var/log/ /tmp/uds-containerd-logs
+          else
+            echo "Container $CONTAINER_NAME is not running. Skipping log copy."
+          fi
+      - description: Dump Node Logs
+        cmd: |
+          docker ps --filter "name=k3d" --format "{{.Names}}" | while read line; do
+            docker logs "$line" 2> /tmp/$line.log
+          done
+      - description: Fix log permissions
+        cmd: |
+          sudo chown $USER /tmp/zarf-*.log || echo ""
+          sudo chown $USER /tmp/uds-*.log || echo ""
+
+  - name: setup-environment
+    description: UDS Environment Setup
+    actions:
+
```

Comment:
Is there a rationale to some tasks having the extra empty lines and some not between action declaratins?

Reply:
realistically i did save-logs first and it was plenty readable and then setup-environment was not